rum-injection-installer/installer/nginx_configurator.go

```diff
@@ -0,0 +1,272 @@
+// Checks that the version is higher than a min version
+func (n *NginxConfigurator) Verify() error {
+	cmd := exec.Command("nginx", "-v")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("nginx is not installed")
+	}
+
+	versionRegex := regexp.MustCompile(`nginx version: nginx/(\d+\.\d+\.\d+)`)
+	matches := versionRegex.FindStringSubmatch(string(output))
+	if len(matches) < 2 {
+		return fmt.Errorf("failed to parse nginx version: %s", output)
+	}
+
+	n.Version = matches[1]
+
+	print("Detected NGINX version: ", n.Version, "\n")
+
+	const minSupportedVersion = "1.22.0"
+	if compareVersions(n.Version, minSupportedVersion) < 0 {
+		return fmt.Errorf("nginx version %s is not supported, must be at least %s", n.Version, minSupportedVersion)
+	}
+
+	return nil
+}
+
+// Reads the config file
+// 1. Checks if a line containing the module already exists
+// 2. Creates a backup of the config file
+// 3. Appends the load_module line to the beginning of the file
+// 4. Inserts the RUM config as early as possible in the http section
+func (n *NginxConfigurator) ModifyConfig(appID, site, clientToken, agentUrl string, sessionSampleRate, sessionReplaySampleRate int) error {
+	configPath, err := n.getConfigLocation()
+	if err != nil {
+		return fmt.Errorf("failed to get config location: %v", err)
+	}
+
+	content, err := os.ReadFile(configPath)
+	if err != nil {
+		return fmt.Errorf("failed to read config file: %v", err)
+	}
+
+	// Check if Datadog module is already loaded
+	if strings.Contains(string(content), "ngx_http_datadog_module.so") {
+		return fmt.Errorf("datadog module is already loaded in the configuration")
+	}
+
+	// Create backup
+	backupPath := configPath + ".backup"
+	if err := os.WriteFile(backupPath, content, 0644); err != nil {
+		return fmt.Errorf("failed to create backup file: %v", err)
+	}
+	fmt.Printf("Backup created: %s\n", backupPath)
+
+	var newContent strings.Builder
+	newContent.WriteString(fmt.Sprintf("load_module %s/ngx_http_datadog_module.so;\n", n.ModulesPath))
+
+	// Find http and bracket { on possibly multiple lines
+	httpRegex := regexp.MustCompile(`(?m)^[\s]*http[\s]*\{`)
+
+	httpBlockStart := httpRegex.FindIndex(content)
+	if httpBlockStart == nil {
+		return fmt.Errorf("could not find http block in nginx config")
+	}
+
+	newContent.Write(content[:httpBlockStart[0]])
+
+	newContent.Write(content[httpBlockStart[0]:httpBlockStart[1]])
+	newContent.WriteString("\n")
+
+	datadogConfig := fmt.Sprintf(`    datadog_agent_url %s;
+    datadog_disable;
+    datadog_rum on;
+    datadog_rum_config "v5" {
+        "applicationId" "%s";
+        "clientToken" "%s";
+        "site" "%s";
+        "sessionSampleRate" %d;
+        "sessionReplaySampleRate" %d;
+    }
+`, agentUrl, appID, clientToken, site, sessionSampleRate, sessionReplaySampleRate)
+	newContent.WriteString(datadogConfig)
+
+	newContent.Write(content[httpBlockStart[1]:])
+
+	// Write the modified content back to the file
```

Comment:
I would feel more confortable to write the modified configuration somewhere on disk, validate it with `nginx -c <CONFIG> -t`, then, swap with the default config if valid, otherwise, abort and and delete the modified configuration.